### ISSUE TITLE
Feature ideas: add translated selections to anki/add word context

### ISSUE THREAD

**Issue by anaxonda** [Ajatt-Tools/anki.koplugin#39]
1) It would be useful to have selections that have been translated also added to anki, with the text selection and the resulting translation.

However, I'm not sure if these are stored somewhere already.

2) Add the context of the word lookup to the anki card (the sentence that contains the looked-up word).

**Comment by anaxonda** [comment]
Oops, my fault for not reading well, I see #2 is already an option.

**Comment by nairyosangha** [comment]
Not sure I understand what you're envisioning for your first suggestion. You want the sentence you looked up to be translated on the fly and inserted on the note as well? KOreader doesn't provide anything like that, you'd need to query some third party service.

**Comment by anaxonda** [comment]
Yeah, I'm referring to the built in translation service, its one of the selection options.

**Comment by nairyosangha** [comment]
Neat, I admit I did not check at all whether KOreader supported that already, this was pretty straightforward: cf58b4209dc7226e16bd0f40243556c1bc8ef2cc 

Could you give it a try and see if anything is missing? It inherits language settings from the settings you define in KOReader itself so it should be pretty simple.
One thing to keep in mind is that this will make syncing your notes even slower because it has to do an extra call for each note now, it's already pretty slow as is, but not much I can do about that.

**Comment by anaxonda** [comment]
Should I expect to see an `add to anki` button when I view a translation? Right now there is none.

**Comment by nairyosangha** [comment]
> Should I expect to see an `add to anki` button when I view a translation? Right now there is none.

No, what I did is update the normal flow, if you look up a word in a dictionary and choose to add it, it will also add a translation of the current sentence, provided you filled in a value for the accompanying `translated_context_field`. You don't get to pick a translation yourself. It'll use the target lang you specify in KOreader's translation settings

**Comment by nairyosangha** [comment]
I took a quick look and adding the 'add to anki' button on the translation window wouldn't work without modifying the KOreader code base itself so I'd rather not try to get that to work

**Comment by nairyosangha** [comment]
Closing this one since this is supported now, feel free to reopen if you have further remarks
